media-proxy/src/shm_memif_common.c

```diff
@@ -196,3 +196,24 @@ int tx_on_disconnect(memif_conn_handle_t conn, void* priv_data)
 
     return 0;
 }
+
+int memif_buffer_alloc_timeout(memif_conn_handle_t conn, uint16_t qid,
+                               memif_buffer_t * bufs, uint16_t count, uint16_t * count_out,
+                               uint32_t size, uint32_t timeout_ms) {
+    uint32_t waited_half_ms = 0;
+    int err;
+
+    while (waited_half_ms < timeout_ms * 2) {
+        err = memif_buffer_alloc(conn, qid, bufs, count, count_out, size);
+        if (err == MEMIF_ERR_NOBUF_RING) {
+            usleep(500);
+            waited_half_ms += 1;
+            continue;
+        }
+        if (err)
+            return err;
+        else
+            return 0;
+    }
+    return MEMIF_ERR_NOBUF_RING;
+}
```

Comment:
```suggestion
int memif_buffer_alloc_timeout(memif_conn_handle_t conn, uint16_t qid,
                               memif_buffer_t * bufs, uint16_t count, uint16_t * count_out,
                               uint32_t size, uint32_t timeout_ms) {
    timeout_ms = timeout_ms*2;
    int err;

    do {
        err = memif_buffer_alloc(conn, qid, bufs, count, count_out, size);
        if (err != MEMIF_ERR_NOBUF_RING) {
            return err;
         }
         usleep(500);
    } while (--timeout_ms)
    return MEMIF_ERR_NOBUF_RING;
}
```

Reply:
Ony asking if this could be like above.

Reply:
You're proposing to invert the logic of exiting the loop. Do you think this is easier to understand the code?

Reply:
Ok you've got the point, this is less readable.